pymeasure/display/widgets/sequencer_widget.py

```diff
@@ -556,13 +557,14 @@ def load_sequence(self, *, filename=None):
         :param filename: Filename (string) of the to-be-loaded file.
         """
         append_flag = False
+
         if (filename is None) or (filename == ''):
             dialog = SequenceDialog()
-            dialog.exec()
-            append_flag = dialog.append_checkbox.checkState() == QtCore.Qt.CheckState.Checked
-            filenames = dialog.selectedFiles()
-            if filenames:
-                filename = filenames[0]
+            if dialog.exec():
```

Comment:
I am not sure what issue this is fixing.
If you hit cancel, `filenames` will be empty so the function will return without any modification.
Right?